python/GangaDirac/Lib/Backends/DiracUtils.py

```diff
@@ -30,27 +30,17 @@ def get_result(command,
 
         try:
             result = execute(command, eval_includes=eval_includes)
-
-            if not result_ok(result):
-                if logger_message is not None:
-                    logger.warning('%s: %s' % (logger_message, str(result)))
-                if exception_message is not None:
-                    logger.warning("Failed to run: %s" % str(command))
-                    logger.warning("includes:\n%s" % str(eval_includes))
-                    logger.warning("Result: '%s'" % str(result))
-                    raise GangaException(exception_message)
-                raise GangaException("Failed to return result of '%s': %s" % (command, result))
-            return result
-        except Exception as x:
-            import time
+        except GangaDiracException as err:
+            logger.error(exception_message)
             logger.debug("Sleeping for 5 additional seconds to reduce possible overloading")
             time.sleep(5.)
             if retries == retry_limit - 1:
                 raise
             retries = retries + 1
-            logger.error("An Error Occured: %s" % str(x))
+            logger.error("An Error Occured: %s" % str(err))
             logger.error("Retrying: %s / %s " % (str(retries + 1), str(retry_limit)))
 
+        return result
```

Comment:
This `return result` should probably be inside the `try` block. Otherwise, if an exception is raised, `result` will not be defined.